### PR TITLE
fix file resolving to work with root-level files

### PR DESCRIPTION
**Pull request by FredKSchott**
This is a small fix related to how URLs are resolved. This was the visible bug that caused all the trouble in https://github.com/Polymer/polymer-build/pull/83.

/cc @justinfagnani @usergenic @rictic

### CODE REVIEW

File: lib/pathresolver.js [hunk 1]
```diff
@@ -86,7 +86,14 @@ PathResolver.prototype = {
     var parsedFrom = url.parse(mainDocUrl);
     var parsedTo = url.parse(absUrl);
     if (parsedFrom.protocol === parsedTo.protocol && parsedFrom.host === parsedTo.host) {
-      var pathname = pathPosix.relative(pathPosix.dirname(parsedFrom.pathname), parsedTo.pathname);
+      var parsedFromDir = pathPosix.dirname(parsedFrom.pathname);
+      // top-level file paths like `index.html` will return the dirname `.`.
+      // To calculate the relative path below, we should resolve against the
+      // root (`/`) instead.
+      if (parsedFromDir === '.') {
```
Comment (usergenic) [inline lib/pathresolver.js]: FWIW, the new v3 branch bundler code works around this with a custom URL relative calculation function that incorporates the logic for isAbsoluteUrl checks etc., so I don't want to get too in depth on this fix-- just trying to understand if there is a case where we expect top-level path not to include the leading '/' -- The tests don't seem to have a case for this, so I just wasn't sure how this branch gets exercised...?  Wouldn't this only happen if we were testing relative imports, not absolute imports?  This PR added '/html/absolute-imports.html'.

I feel like I am missing something... HALP!

File: lib/pathresolver.js [hunk 2]
```diff
@@ -86,7 +86,14 @@ PathResolver.prototype = {
     var parsedFrom = url.parse(mainDocUrl);
     var parsedTo = url.parse(absUrl);
     if (parsedFrom.protocol === parsedTo.protocol && parsedFrom.host === parsedTo.host) {
-      var pathname = pathPosix.relative(pathPosix.dirname(parsedFrom.pathname), parsedTo.pathname);
+      var parsedFromDir = pathPosix.dirname(parsedFrom.pathname);
+      // top-level file paths like `index.html` will return the dirname `.`.
```
Comment (justinfagnani) [inline lib/pathresolver.js]: I find this comment a little hard to grok. What's a top-level path, and where do we get it?

File: lib/pathresolver.js (hunk 1 above)
Comment (usergenic) [inline lib/pathresolver.js]: For reference, the new relativeUrl function in v3 for reference.  https://github.com/Polymer/vulcanize/blob/v3/src/url-utils.ts#L48
https://github.com/Polymer/vulcanize/blob/v3/src/test/url-utils_test.ts#L67

None of tests address a case where pathname of the "from" URL isn't absolute, but I just tried calling the new relativeUrl with args 'index.html', 'some-other-thing.html' and it did the right thing, so I think we'll not see a regression in build when we switch to v3.

File: lib/pathresolver.js [hunk 3]
```diff
@@ -86,7 +86,14 @@ PathResolver.prototype = {
...
+      // To calculate the relative path below, we should resolve against the
+      // root (`/`) instead.
+      if (parsedFromDir === '.') {
+        parsedFromDir = path.sep;
```
Comment (FredKSchott) [inline lib/pathresolver.js]: right, good catch

**Comment by FredKSchott** [comment]
No longer needed for polymer-build thanks to https://github.com/Polymer/polymer-build/pull/83. 

This may still be a problem for other projects, but given the in-progress v3 branch where this is already fixed, I don't think it's worth potentially breaking things for others now.

**Pull request closed without merging**